src/app/pages/marker-generator/marker-generator.component.ts

```diff
@@ -0,0 +1,228 @@
+    pageOrientationControl: new FormControl({ pageOrientation: 'portrait', numPerPage: 1 }, Validators.required),
+    includeFlowersControl: new FormControl(true, Validators.required),
+    includeNestsControl: new FormControl(true, Validators.required),
+    numNestsControl: new FormControl(30, [
+      Validators.required,
+      Validators.min(1),
+      Validators.max(this.maxNumNests)
+    ]),
+  },
+    {
+      validators: (fg: FormGroup) =>
+        (fg.controls.includeFlowersControl.value || fg.controls.includeNestsControl.value) ? null : { 'no-flowers-or-nests': true }
+    });
+
+  compareOrientationAndNumPerPage(a: OrientationAndNumPerPage, b: OrientationAndNumPerPage) {
+    return a.pageOrientation === b.pageOrientation && a.numPerPage === b.numPerPage;
+  }
+
+  constructor(public sanitizer: DomSanitizer) { }
+
+  ngOnInit(): void {
+
+  }
+
+
+  getBarcode(value: number) {
+    return new CustomBarcodeMarkerGenerator(BARCODE_TYPE, value);
+  }
+
+  flowerPages(markerSize: number): Page[] {
+    return rangeArray(MIN_FLOWER_MARKER, MAX_FLOWER_MARKER).map(val => ({
+      value: `${val}`,
+      type: 'Flower',
+      backgroundSVG: this.getBarcode(val).asSVGWithSize(markerSize)
+    }));
+  }
+
+  nestPages(nestCount: number, markerSize: number): Page[] {
+    if (nestCount === 0) {
+      return [];
+    }
+    return rangeArray(MIN_NEST_MARKER, MIN_NEST_MARKER + nestCount - 1).map(val => ({
+      value: `${val}`,
+      type: 'Nest',
+      backgroundSVG: this.getBarcode(val).asSVGWithSize(markerSize)
+    }));
+  }
+
+
+  pdfFromPages(
+    pages: Page[],
+    svgHeight: number, // height in points
+    pageSize: PageSize = 'LETTER',
+    { pageOrientation, numPerPage }: OrientationAndNumPerPage = { pageOrientation: 'portrait', numPerPage: 1 }
+  ): pdfMake.TCreatedPdf {
+
+
+    const markerColumns: Column[][] = pages.map(p => ([
+      {
+        // nesting columns here to allow the outer column to have width '*'
+        // while allowing the SVG to have a valid width.
+        columns: [
+          {
+            svg: beeSVG,
+            height: 40,
+            width: 40,
+            relativePosition: { x: 0, y: 0 }
+          }
+        ],
+        width: '*'
+      },
+      {
+        width: 'auto',
+        stack: [
+          p.type,
+          {
+            text: p.value,
+            fontSize: 24
+          }
+        ],
+        alignment: 'center'
+      },
+      {
+        width: '*',
+        stack: [
+          'Buzz About',
+          `Markers v${MARKERS_VERSION}`
+        ],
+        alignment: 'right'
+      }
+    ]));
+
+    const columnPages = chunk(markerColumns, numPerPage);
+
+    const backgroundSVGPages = chunk(pages.map(p => p.backgroundSVG), numPerPage);
+
+    if (numPerPage > 1 && pages.length % 2) {
+      columnPages.push([...columnPages.pop(), []]);
+      backgroundSVGPages.push([...backgroundSVGPages.pop(), blankSVG]);
```

Comment:
This logic is actually flawed for `numPerPage > 2` when `pages.length % numPerPage < numPerPage - 1). It should really add an additional `numPerPage - (pages.length % numPerPage)` blank sections. For now this works since the only `numPerpage` possible are 1 and 2.

Reply:
Yeah, I saw that too! :) I also figured it was fine.

Reply:
I have just corrected this in 6e0739bc6281e079807dcb0f575ed8d2dd9caab5